solcjs

```diff
@@ -14,12 +14,26 @@ var smtsolver = require('./smtsolver.js');
 process.removeAllListeners('uncaughtException');
 var commander = require('commander');
 
+const parseOptimizeRuns = function(value) {
+  const parsedValue = parseInt(value, 10);
+  if (isNaN(parsedValue)) {
+    return value;
```

Comment:
The official documentation suggests to raise an error here:
```suggestion
    throw new commander.InvalidArgumentError('Not a number.');
```

Reply:
I tested it locally and I think it's fine. It does report errors for non-int values properly. If the value is not an int, it just puts it into JSON and lets it do any further validation.

Actually, one thing I'm not certain about is if it matches solc's full integer range. In JS there are only floats so some larger ints that the compiler would accept might be passed to it as strings. Or worse - be rounded? @rorye, I think we should have a test for the largest possible int value.

> why does it work with the current code (it checks that the type of the input and the output differs?)

I don't understand this question. AFAIK commander does not check the type at all, which is why it's being converted to int here.

Reply:
The code above does not exceptions just returns the input, yet [this test case](https://github.com/ethereum/solc-js/pull/546/files#diff-aae216853526bb5eb76ffeff66cc7f56184296adde901fd9d48c08e925723a8aR49) seemingly works. I'm not sure how.

Reply:
It works because it's `compile()` that raises the exception. The code here just returns non-int values unchanged which makes the compiler binary report an error later.

Reply:
Ah okay, I guess it is fine using that but I'd document it. Alternatively just do the `throw` as I suggested, that is in line with the CLI library.

Reply:
@axic - Added the throw as suggested and the corresponding catch. I could not use `commander.InvalidArgumentError` as it resulted in 

```
Error: "commander.InvalidArgumentError is not a constructor
```
Perhaps the version of commander we are using is a little old.

Reply:
@rorye It is old. I needed to upgrade it for my own PR too. I have extracted the change that upgrades it into a separate PR (#547). You can rebase yours on it.